File: SwiftProject/SwiftProject/Kex_2.swift
//
//  Kex_2.swift
//  SwiftProject
//
//  Created by 周哲輝 on 2019/06/24.
//  Copyright © 2019 周哲輝. All rights reserved.
//

import Foundation

public class Kex_2 {
    public static func Execute() {
        print("冒険が今始まる！")
        var player = Player()
        var continueGame = 1
        var kills = 0
        
        while continueGame != 0 {
            print("現HP：\(player.Hp)")
            print("奥に進みますか＞？（１：奥に進む　０：帰る）＞", terminator: "")
            continueGame = Int(readLine()!)!
            if continueGame == 1 {
                let type = EnemyType(rawValue: Int.random(in: 0...2))!
                var enemy = CreateEnemy(type: type)
                let result = AttackEnemy(player: &player, enemy: &enemy)
                kills += 1
                
                if !result {
                    break
                }
            }
        }
        print("リ○ミト！\n")
        print("戦闘回数：\(kills)回　残りHP：\(player.Hp)")
    }
}

File: rust/src/questions/kex_2.rs
use crate::player::Player;
use crate::read_user_input::read_user_input;
use std::io::Write;
use rand::Rng;
use crate::kex_2_enemy::{create_enemy, EnemyType};
use crate::kex_2_battle::attack_enemy;

pub struct Kex_2 {}
impl Kex_2 {
    pub fn execute() {
        println!("冒険が今始まる！");
        let mut player = Player::new();
        let mut continue_game = 1;
        let mut kills = 0;

        while continue_game != 0 {
            println!("現HP：{}", player.hp);
            print!("奥に進みますか？（１：奥に進む　０．帰る）＞");
            std::io::stdout().flush().unwrap();
            let mut reader = String::new();
            continue_game = match read_user_input(&mut reader) {
                Ok(x) => x,
                Err(ex) => {
                    eprintln!("エラーが発生しました：{}", ex.to_string());
                    return;
                }
            };
            if continue_game == 1 {
                let lv = rand::thread_rng().gen_range(0, 3);
                let mut enemy = create_enemy(EnemyType::from(lv));
                let result = attack_enemy(&mut player, &mut enemy);
                kills += 1;
                if !result {
                    break;
                }
            }
        }
        println!("リ〇ミト！\n");
        println!("戦闘回数：{}回　残りHP：{}", kills, player.hp);
    }
}